packages/react-components/react-motion-components-preview/library/src/components/Fade/Fade.test.tsx

```diff
@@ -0,0 +1,131 @@
+
+    rerender(<FadeSnappy visible={true}>{testElement}</FadeSnappy>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationFast, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Exaggerated variant of Fade componentwith correct opacity keyframes, duration and easing', () => {
+    const { rerender } = render(<FadeExaggerated visible={false}>{testElement}</FadeExaggerated>);
+
+    rerender(<FadeExaggerated visible={true}>{testElement}</FadeExaggerated>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationGentle, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render child element with overriding styles and still apply Fade animations', () => {
+    const styledElement = <div style={{ opacity: 0.5 }}>Test</div>;
+
+    const { rerender } = render(<Fade visible={false}>{styledElement}</Fade>);
+
+    rerender(<Fade visible={true}>{styledElement}</Fade>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Fade with parent styles that override Fade opacity and still apply Fade animations (visible=false -> true -> false)', () => {
+    const parentStyle = `
+      .parentOverride, .parentOverride > .testContainer {
+        opacity: 0.5 !important; /* Override any opacity applied by Fade */
+      }
+    `;
+
+    const styleElement = document.createElement('style');
+    styleElement.innerHTML = parentStyle;
+    document.head.appendChild(styleElement);
+
+    const { rerender, container } = render(
+      <div className="parentOverride">
+        <Fade visible={false}>{testElement}</Fade>
+      </div>,
+    );
+
+    const parentElement = container.firstChild as HTMLElement;
+
+    setTimeout(() => {
+      const parentComputedStyles = getComputedStyle(parentElement);
+      expect(parentComputedStyles.opacity).toBe('0.5');
+    }, 0);
+
+    rerender(
+      <div className="parentOverride">
+        <Fade visible={true}>{testElement}</Fade>
+      </div>,
+    );
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+
+    rerender(
+      <div className="parentOverride">
+        <Fade visible={false}>{testElement}</Fade>
+      </div>,
+    );
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 1 }, { opacity: 0 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+
+    document.head.removeChild(styleElement);
```

Comment:
This style is being removed before the `setTimeout` code executes, so the `getComputedStyle` assertion doesn't work. This line can be moved into the `setTimeout`.